vktrace/vktrace_replay/vkreplay_vkreplay.cpp

```diff
@@ -558,9 +558,27 @@ VkResult vkReplay::manually_replay_vkEnumeratePhysicalDevices(packet_vkEnumerate
         } else if (pDevices != NULL) {
             vktrace_LogVerbose("Enumerated %d physical devices in the system.", deviceCount);
         }
-        // TODO handle enumeration results in a different order from trace to replay
-        for (uint32_t i = 0; i < deviceCount; i++) {
-            if (pDevices != NULL) {
+        if (pDevices != NULL) {
+            for (uint32_t i = 0; i < deviceCount; i++) {
```

Comment:
This change is intended to handle the following case, which introduces a device count mismatch between trace and replay:
* The trace layer intercepted vkEnumeratePhysicalDevices before the Optimus layer and received: trace_devices = { intel_gpu, nvidia_gpu }
* vkreplay received a list of devices from vkEnumeratePhysicalDevices after it has been filtered by the Optimus layer: replay_devices = { nvidia_gpu }
* vkreplay needs to map trace_devices[1] to replay_devices[0]

To map each physical device from the trace file to a replay device, the loop counter should use *(pPacket->pPhysicalDeviceCount).  A mismatch between the trace and replay physical device counts will also need to be handled.

I think you want something like this:
```
for(i = 0; i < trace_device_count; i++)
    if ((i < replay_device_count) && (trace_device_id[i] == replay_device_id[i]))
        // Exact match
    else
        // Search for a match in the list of available physical devices
        for (j = 0; j < replay_device_count; j++)
                if (trace_device_id[i] == replay_device_id[j])
                    // Map i to j
```